copy/copy.go

```diff
@@ -489,8 +497,20 @@ func (ic *imageCopier) copyBlobFromStream(srcStream io.Reader, srcInfo types.Blo
 		inputInfo.Size = -1
 	}
 
+	var stream io.Reader
+
 	// === Finally, send the layer stream to dest.
-	uploadedInfo, err := ic.dest.PutBlob(destStream, inputInfo)
+	// the pipe is here to facilitate the stream copier which will act as an
+	// intermediary to signal progress information to any registered channel.
+	if ic.signal != nil && ic.signalInterval > 0 {
+		r, w := io.Pipe()
```

Comment:
`defer r.Close()` somewhere, otherwise we could exit with an unreferenced unclosed `PipeReader` and a `PipeWriter` still blocked on trying to write to it.